### PR TITLE
feat: cli initial setup 

### PR DESCRIPTION
**Pull request by ArzelaAscoIi**
### Related Issues

- add cli commands for uploading files

### changes
- add cli for uploading paths and files 

### Tests
- unit tests (no need to have integration tests since this is already integration tested) 

### Notes 
I did NOT add check if the package.toml works as expected and users can run `deepset_cloud_cli <command>`

### Checklist

- [ ] I have updated the referenced issue with new insights and changes
- [ ] If this is a code change, I have added unit tests
- [ ] I've used the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title
- [ ] I updated the docstrings
- [ ] If this is a code change, I added meaningful logs and prepared Datadog visualizations and alerts

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,13 +27,17 @@ dependencies = [
   "structlog==23.1.0",
   "httpx==0.24.0",
   "python-dotenv==1.0.0",
+  "typer==0.9.0",
```
Comment (rjanjua) [inline pyproject.toml:30]: for some reason the lint step in the CI is complaining that it could not import typer

**Review by rjanjua — CHANGES_REQUESTED** [review]
Looks really good. The CI isn't completely working though. I want to deploy this to the test PyPi so that we can test if it works as intended. I am not sure that everything is set up correctly for us to use it as described in the README

File: pyproject.toml (hunk 1 above)
Comment (ArzelaAscoIi) [reply pyproject.toml:30]: i know. I guess its somehow related to hatch and caching.

**Comment by ArzelaAscoIi** [comment]
>Looks really good. The CI isn't completely working though. I want to deploy this to the test PyPi so that we can test if it works as intended. I am not sure that everything is set up correctly for us to use it as described in the README

yes. i think there is some path variable missing

**Review by rjanjua — APPROVED** [review]
🚀

**Pull request merged**